src/http/formdata/parser.cr

```diff
@@ -0,0 +1,142 @@
+module HTTP::FormData
+  class ParseException < Exception
+  end
+
+  # Parses a multipart/form-data message. Callbacks are used to process files
+  # and fields.
+
+  # ### Example
+  #
+  # ```
+  # form_data = "--aA40\r\nContent-Disposition: form-data; name=\"field1\"\r\n\r\nfield data\r\n--aA40--"
+  # parser = HTTP::FormData::Parser.new(MemoryIO.new(form_data), "aA40")
+  #
+  # parser.field("field1") do |data|
+  #   data # => "field data"
+  # end
+  #
+  # parser.run
+  # ```
+  class Parser
```

Comment:
I think we need a callback to handle all fields, regardless of the name. It might very well be desirable to handle inputs with unknown names.